config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -244,14 +257,12 @@ type AlertmanagerCustomConfig struct {
 	// +listMapKey=whenUnsatisfiable
 	// +optional
 	TopologySpreadConstraints []v1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
-	// volumeClaimTemplate Defines persistent storage for Alertmanager. Use this setting to
-	// configure the persistent volume claim, including storage class, volume
-	// size, and name.
+	// volumeClaimTemplate defines persistent storage for Alertmanager. Use this setting to
+	// configure the persistent volume claim, including storage class and volume size.
 	// If omitted, the Pod uses ephemeral storage and alert data will not persist
 	// across restarts.
-	// This field is optional.
 	// +optional
-	VolumeClaimTemplate *v1.PersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
+	VolumeClaimTemplate VolumeClaimConfig `json:"volumeClaimTemplate,omitempty,omitzero"`
```

Comment:
Be cautious of changing this.

If this changes the serialization of the type you'll need to upversion this to v1alpha2.

I had forgotten that `*v1.PersistentVolumeClaim` had been used in the past so it may be OK to stay consistent. Before promoting to a v1 though I'd recommend converting to a custom type so you can enforce better validations at admission time.